common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
```

Comment:
@gianm What do you suggest the API for `SecretProvider` Interface look like, should this be same as PasswordProvider as per the changes in this PR, meaning are you just suggesting this design to decouple secret fetching logic from `MetadataStorageConnectorConfig` creation or something else ?
@drcrallen I'll do another PR for these changes